File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/domain/entity/LangEntity.java
package ru.spring5.elena.librarymanager.domain.entity;

import lombok.*;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "LANGS")
@Getter
@NoArgsConstructor
//@RequiredArgsConstructor
@Setter
public class LangEntity {
    @Id
    @Column(name = "LANG_ID")
    @NotNull
    @Setter(AccessLevel.NONE)
    private Long langId;

    @Version
    @Column(name = "VERSION")
    private int version; //версия, инкрементится при редактировании записи

    @Column(name = "TITLE")
    private String title; //
    
    @OneToMany(mappedBy = "lang", cascade = CascadeType.ALL, orphanRemoval = true)
    @Setter(AccessLevel.NONE)
    private Set<EditionEntity> editions = new HashSet<EditionEntity>();
    
    @OneToMany(mappedBy = "publLang", cascade = CascadeType.ALL, orphanRemoval = true)
    @Setter(AccessLevel.NONE)
    private Set<EditionEntity> publLangEditions = new HashSet<EditionEntity>();
}

File: librarymanager-be/src/main/kotlin/ru/elenacherev/librarymanager/domain/entity/LangEntity.kt
package ru.elenacherev.librarymanager.domain.entity

import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table
import javax.persistence.Version
import javax.validation.constraints.NotNull

@Entity
@Table(name = "LANGS")
class LangEntity(
    @Id
    @Column(name = "LANG_ID")
    @get: NotNull
    var langId: Long? = null,

    //версия, инкрементится при редактировании записи
    @Version
    @Column(name = "VERSION")
    var version: Int = 0,

    @Column(name = "TITLE")
    var title: String,

    @OneToMany(mappedBy = "lang", cascade = [CascadeType.ALL], orphanRemoval = true)
    val editions: MutableSet<EditionEntity> = mutableSetOf(),

    @OneToMany(mappedBy = "publLang", cascade = [CascadeType.ALL], orphanRemoval = true)
    val publLangEditions: MutableSet<EditionEntity> = mutableSetOf<EditionEntity>()
)